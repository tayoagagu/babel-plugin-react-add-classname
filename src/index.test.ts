import pluginTester from 'babel-plugin-tester'
import { getSnapshotSerializer } from 'string-snapshot-serializer'
import plugin from '.'

expect.addSnapshotSerializer(getSnapshotSerializer())

pluginTester({
  title: 'no options',
  plugin,
  snapshot: true,
  tests: [
    {
      title: 'do nothing',
      code: `
  const hello = () => (
    <div className="test-hello"></div>
  )
`,
    },
  ],
})

pluginTester({
  title: 'with option',
  plugin,
  snapshot: true,
  pluginOptions: { name: 'classname-to-add' },
  tests: [
    {
      title: 'with class name',
      code: `
  const hello = () => (
    <div className="babel-a hello test-b"></div>
  )
`,
    },
    {
      title: 'with class name that already exists',
      code: `
  const hello = () => (
    <div className="babel-a hello test-b classname-to-add"></div>
  )
`,
    },
  ],
})
