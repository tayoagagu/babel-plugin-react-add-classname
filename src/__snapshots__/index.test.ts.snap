// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no options do nothing: do nothing 1`] = `

const hello = () => (
  <div className="test-hello"></div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = () => <div className="test-hello"></div>;

`;

exports[`with option with class name that already exists: with class name that already exists 1`] = `

const hello = () => (
  <div className="babel-a hello test-b classname-to-add"></div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = () => <div className="babel-a hello test-b classname-to-add"></div>;

`;

exports[`with option with class name: with class name 1`] = `

const hello = () => (
  <div className="babel-a hello test-b"></div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = () => <div className="babel-a hello test-b classname-to-add"></div>;

`;
